.github/workflows/compilation-checks.yml

```
@@ -0,0 +1,31 @@
+# Copyright 2020, Data61, CSIRO (ABN 41 687 119 230)
+#
+# SPDX-License-Identifier: BSD-2-Clause
+
+# Compilation actions to run on pull requests
+
+name: Compilation checks
+
+on: [pull_request]
```

```suggestion
on:
  push:
    branches:
      - master
  pull_request:
```

I'd run these on push to master as well. Theoretically it should already be implied that this always works on master if we check on every PR, but it's nice to be able to look at just the branch and directly see the CI status.